### PR TITLE
Speeding up SDK dependencies download

### PR DESCRIPTION
**Pull request by wmathurin**
Using a shallow git clone instead of npm dependencies (which were also getting submodules)

### CODE REVIEW

File: ReactNativeTemplate/ios/Podfile [hunk 1]
```diff
@@ -1,5 +1,4 @@
-# Uncomment this line to define a global platform for your project
-# platform :ios, '9.0'
+platform :ios, '10.0'
```
Comment (bhariharan) [inline ReactNativeTemplate/ios/Podfile]: Shouldn't this be `9.0`? That's the minimum we support, not `10.0`.

File: iOSNativeSwiftTemplate/Podfile [hunk 2]
```diff
@@ -1,5 +1,4 @@
-# Uncomment this line to define a global platform for your project
-# platform :ios, '9.0'
+platform :ios, '10.0'
```
Comment (bhariharan) [inline iOSNativeSwiftTemplate/Podfile]: Should be `9.0`.

File: iOSNativeTemplate/Podfile [hunk 3]
```diff
@@ -1,5 +1,4 @@
-# Uncomment this line to define a global platform for your project
-# platform :ios, '9.0'
+platform :ios, '10.0'
```
Comment (bhariharan) [inline iOSNativeTemplate/Podfile]: Should be `9.0`.

File: iOSNativeTemplate/install.js [hunk 4]
```diff
@@ -1,7 +1,20 @@
...
 var execSync = require('child_process').execSync;
-console.log('Installing npm dependencies');
-execSync('npm install', {stdio:[0,1,2]});
+var path = require('path');
+var fs = require('fs');
+
+console.log('Installing sdk dependencies');
+for (var sdkDependency in packageJson.sdkDependencies) {
+    var repoUrlWithBranch = packageJson.sdkDependencies[sdkDependency];
+    var parts = repoUrlWithBranch.split('#'), repoUrl = parts[0], branch = parts.length > 1 ? parts[1] : 'master';
```
Comment (bhariharan) [inline iOSNativeTemplate/install.js]: Why `master`? Shouldn't we use tag? What exactly is this doing?

Comment (wmathurin) [inline iOSNativeTemplate/install.js]: I read the SDK repo URL from the package.json. Then I do a git clone (shallow). But first I need to extract the tag or branch from the URL. Repo#branch is a syntax understood by npm not git. If the URL doesn't have a tag or branch specified then it means one wants master. I could also drop the branch option from the git clone command in that case.

File: ReactNativeTemplate/ios/Podfile (hunk 1 above)
Comment (wmathurin) [inline ReactNativeTemplate/ios/Podfile]: I will do that - devs could change the podfile and rerun pod update it they want 9.0.

**Comment by wmathurin** [comment]
Target iOS 9 by default: https://github.com/forcedotcom/SalesforceMobileSDK-Templates/pull/41/commits/93c3eba410ac16b119f5d3cdd1235925cfcdea6b

File: iOSNativeTemplate/install.js (hunk 4 above)
Comment (bhariharan) [inline iOSNativeTemplate/install.js]: Sounds good.

**Comment by wmathurin** [comment]
Unfortunately we need to get the submodules: some of them (e.g. fmdb) are needed to compile the sdk - some are useless for native/react native apps (e.g. cordova, react-native, shared).
I'll send a commit with the change to the install.js that need it.

**Comment by wmathurin** [comment]
The connection can get slow here. It makes a big difference. Not having to download the cordova/react-native submodules is a time saver.

**Pull request merged**